File: TrendAnalyzerServer/src/main/java/vorotilenko/trendanalyzerserver/server/ClientMessage.java
package vorotilenko.trendanalyzerserver.server;

/**
 * Class for parsing JSON messages which are sent from client to {@link WSServerEndpoint}.
 */
public class ClientMessage {
    /**
     * One of the {@link ClientMessageTypes}
     */
    public int messageType;
    /**
     * JSON-encoded message wrapped in the {@link ClientMessage}
     */
    public String message;
}

File: TrendAnalyzer/app/src/main/java/vorotilenko/trendanalyzer/serverinteraction/ClientMessage.kt
package vorotilenko.trendanalyzer.serverinteraction

/**
 * Class for parsing JSON messages which are sent from [WSClientEndpoint] to server.
 */
data class ClientMessage(
    /**
     * One of the [ClientMessageTypes]
     */
    val messageType: Int,
    /**
     * JSON-encoded message wrapped in the [ClientMessage]
     */
    val message: String
)